### ISSUE TITLE
Coloured labelling

### ISSUE THREAD

**Issue by 1337ingDisorder** [mate-desktop/caja#270]
It would be excellent if I could right-click a file, choose the "Labels" sub-menu from the right-click context menu, and then set a coloured label from the choices in the "Labels" sub-menu.

This is essentially modelled after the coloured labels one can apply to e-mails in Thunderbird.

One example of this feature's usefulness would be in a folder with a full season of TV episodes -- during a marathon I could mark which episode I left off on when I have to close caja or when I want to navigate to another directory without leaving the current tab or window process running in the background.

Another example would be in a directory with numerous files, some of which were mainly relating to person-or-project A and some of which were mainly relating to person-or-project B, but all of which are categorized together in the main shared directory that both people or project teams could access.

If this feature does get picked up, it'd be extra nifty to have an option at the bottom of the colour-label sub-menu to configure the colours/labels, so the user could have as many different labels as they want and could specify the shades/hues they'd like for their various labels.

Cheers and thanks for reading,
1337ingDisorder
##

**Comment by flexiondotorg** [comment]
This has been implemented via a community contribute extension:

  * http://foldercolor.tuxfamily.org/

**Comment by 1337ingDisorder** [comment]
Just tried the new extension and it seems to work great for the most part :)

It doesn't seem to apply any of the colour labels for me, using Linux 
Mint 18 MATE x64 with Caja set to "details" view mode. When I set a 
colour label it does change the folder's modify timestamp, but it still 
displays the default coloured icon for the desktop theme I have selected 
(which I have set as the basic "GNOME" icon theme using the Appearance 
control panel applet).

However when I apply the "Finished" folder tag it does change the icon 
to show the little green check mark on the icon, which is all I need to 
mark a folder with a video file as having been watched.

So I'm a happy camper! Thanks for implementing this :)

Next question: is there a way to set a folder's tag/colour via 
command-line? It would be nice to be able to automate the 
marking-finished action.

Either way, thanks again! :)
